ledger/participant-integration-api/src/main/scala/platform/store/dao/HikariJdbcConnectionProvider.scala

```diff
@@ -56,12 +58,19 @@ object DataSourceConnectionProvider {
 
       val checkHealth = new TimerTask {
         override def run(): Unit = {
-          try {
-            dataSource.getConnection().close()
-            transientFailureCount.set(0)
-          } catch {
-            case _: SQLTransientConnectionException =>
-              val _ = transientFailureCount.incrementAndGet()
+          LoggingContext.newLoggingContext { implicit loggingContext =>
+            try {
+              dataSource.getConnection().close()
+              transientFailureCount.set(0)
+            } catch {
+              case _: SQLTransientConnectionException =>
+                val _ = transientFailureCount.incrementAndGet()
+              case _: Throwable =>
+                val count = transientFailureCount.incrementAndGet()
+                if (count == 1)
+                  logger.info("Hikari connection health check failed unexpectedly")
```

Comment:
Is it on purpose that if the first exception is `SQLTransientConnectionException` then we'll never log?

Reply:
You are perfectly right. Why didn't I think of it?